File: express/models/order.js
const { Model, DataTypes } = require("sequelize");
const sequelize = require("../config/database");

class Order extends Model {
  static associate(models) {
    Order.belongsTo(models.User, {
      foreignKey: "user_id",
    });

    Order.belongsToMany(models.Product, {
      through: models.OrderProduct,
      as: "orders",
      foreignKey: "order_id",
      otherKey: "product_id",
    });
  }
}

Order.init(
  {
    user_id: {
      type: DataTypes.BIGINT,
      allowNull: false,
    },
    order_date: {
      type: DataTypes.DATE,
      allowNull: false,
      defaultValue: DataTypes.NOW,
    },
    status: {
      type: DataTypes.STRING,
      defaultValue: "Pending",
      allowNull: false,
    },
  },
  {
    sequelize,
    modelName: "Order",
    tableName: "orders",
    timestamps: false,
  }
);

module.exports = Order;

File: asp/performance_asp/performance_asp/Models/Order.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace performance_asp.Models
{
    public class Order
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [ForeignKey("User")]
        [Column("user_id")]
        public int UserId { get; set; }

        [Required]
        [Column("order_date")]
        public DateTime OrderDate { get; set; }

        [Required]
        [DefaultValue("Pending")]
        [Column("status")]
        public string Status { get; set; }

        public virtual ICollection<OrderProduct> OrderProducts { get; set; }
    }
}
